stablehlo/transforms/StablehloRefineShapes.cpp

```diff
@@ -215,6 +215,18 @@ struct EvalConvertOpPattern : public OpRewritePattern<ConvertOp> {
   }
 };
 
+struct EvalDivOpPattern : public OpRewritePattern<DivOp> {
+  using OpRewritePattern::OpRewritePattern;
+  LogicalResult matchAndRewrite(DivOp op,
+                                PatternRewriter& rewriter) const override {
+    auto resultType = op.getResult().getType().cast<ShapedType>();
+    auto isResultUnsigned = resultType.getElementType().isUnsignedInteger();
+    return evalBinary(rewriter, op, [&](APInt lhs, APInt rhs) {
+      return isResultUnsigned ? lhs.udiv(rhs) : lhs.sdiv(rhs);
```

Comment:
Is this a safe optimization to make always? It looks like we're using the same functions in the reference interpreter, not sure if the "upcast to double" is important or an implementation detail:
https://github.com/openxla/stablehlo/pull/1008/files#diff-c2f81cc25e3bff18b8f8ab80deabf5e1c02ee728c3902b5e99ff0505620d27c7R260

Reply:
Just to add one detail: 

For integer types, the reference implementation PR uses  `udiv` or `sdiv` without "upcast to double". The mapper function name "mapWithUpcastToDouble" is a bit misleading in that sense.

Reply:
I think so, yes, given that it conforms to the documented semantics of DivideOp. The "upcast to double" part in the reference implementation only applies to the floating-point part of interpreting the op.